### PR TITLE
[MI-3836] Integrate API to unlink channels and websocket event

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary

- Integrated API to unlink channels
- Added logic to refetch linked channels in rhs when unlinking channels through slash commands

#### Screenshots
image
image
image

### CODE REVIEW

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 1]
```diff
@@ -1,60 +1,128 @@
-import React from 'react';
-
-import {Button, Icon as UILibIcon, Tooltip} from '@brightscout/mattermost-ui-library';
+import React, {useState} from 'react';
+import {useDispatch} from 'react-redux';
 
+import {Button, Icon as UILibIcon, Tooltip, LinearProgress, Dialog} from '@brightscout/mattermost-ui-library';
```
Comment (SaurabhSharma-884) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:4]: Are we using Dialog from ui-lib directly ?

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 2]
```diff
@@ -1,60 +1,128 @@
...
+                destructive={true}
+                primaryButtonText='Unlink channels'
+                secondaryButtonText='Cancel'
+                onSubmitHandler={unlinkChannel}
+                onCloseHandler={() => setShowUnlinkDialog(false)}
+            >
+                {isLoading && <LinearProgress className='absolute w-full left-0 top-62'/>}
+                <span>{'Are you sure you want to unlink the '}<b>{mattermostChannelName}</b>{' and '} <b>{msTeamsChannelName}</b>{' channels? Messages will no longer by synced.'}</span>
+            </Dialog>
+            <Dialog
```
Comment (SaurabhSharma-884) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:114]: Screenshot from 2023-12-14 09-43-20
UI is breaking. Please Check

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 3]
```diff
@@ -78,6 +77,7 @@ export const Rhs = () => {
 
     const {presentInWhitelist} = data as WhitelistUserResponse;
     const {data: linkedChannels, isLoading} = getApiState(pluginApiServiceConfigs.getLinkedChannels.apiServiceName, getLinkedChannelsParams as SearchLinkedChannelParams);
+    const {isLoading: isUserDisconnectingLoading} = getApiState(pluginApiServiceConfigs.disconnectUser.apiServiceName);
```
Comment (SaurabhSharma-884) [inline webapp/src/containers/Rhs/Rhs.container.tsx:80]: `isUserDisconnecting` or `isUserDisconnectLoading`

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 4]
```diff
@@ -332,9 +321,21 @@ export const Rhs = () => {
                         )}
                     </>
                 )}
+                <Dialog
+                    show={showDisconnectDialog}
```
Comment (SaurabhSharma-884) [inline webapp/src/containers/Rhs/Rhs.container.tsx:325]: Screenshot from 2023-12-14 09-53-10
Please check

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 5]
```diff
@@ -1,60 +1,128 @@
...
+                show={showUnlinkDialog}
+                title='Unlink channels'
+                destructive={true}
+                primaryButtonText='Unlink channels'
+                secondaryButtonText='Cancel'
+                onSubmitHandler={unlinkChannel}
+                onCloseHandler={() => setShowUnlinkDialog(false)}
+            >
+                {isLoading && <LinearProgress className='absolute w-full left-0 top-62'/>}
+                <span>{'Are you sure you want to unlink the '}<b>{mattermostChannelName}</b>{' and '} <b>{msTeamsChannelName}</b>{' channels? Messages will no longer by synced.'}</span>
```
Comment (SaurabhSharma-884) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:112]:
```suggestion
                <span>{'Are you sure you want to unlink the '}<b>{mattermostChannelName}</b>{' and '} <b>{msTeamsChannelName}</b>{' channels? Messages will no longer be synced.'}</span>
```

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 6]
```diff
@@ -1,60 +1,128 @@
...
-                </Tooltip>
-                <Tooltip
-                    placement='left'
-                    text={msTeamsTeamName}
-                >
-                    <h5 className='my-0 opacity-6 msteams-linked-channel__body-values'>{msTeamsTeamName}</h5>
-                </Tooltip>
+            <div className='d-flex flex-column gap-6 msteams-linked-channel__body'>
+                <div className='d-flex gap-8 align-items-center'>
+                    {mattermostChannelType === MMConstants.PRIVATE_CHANNEL ? <Icon iconName='lock'/> : <Icon iconName='globe'/>}
```
Comment (SaurabhSharma-884) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:58]:
```suggestion
                    <Icon iconName={mattermostChannelType === MMConstants.PRIVATE_CHANNEL ? 'lock' : 'globe'}/>
```
Check if this works

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 7]
```diff
@@ -1,60 +1,128 @@
...
+
+    // Show unlink and retry dialog component
+    const [showUnlinkDialog, setShowUnlinkDialog] = useState(false);
+    const [showRetryDialog, setShowRetryDialog] = useState(false);
+
+    const {makeApiRequestWithCompletionStatus, getApiState} = usePluginApi();
+    const dispatch = useDispatch();
+    const showAlert = useAlert();
+
+    const {isLoading} = getApiState(pluginApiServiceConfigs.unlinkChannel.apiServiceName, unlinkChannelParams as UnlinkChannelParams);
```
Comment (SaurabhSharma-884) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:29]:
```suggestion
    const {isLoading: isUnlinkChannelsLoading} = getApiState(pluginApiServiceConfigs.unlinkChannel.apiServiceName, unlinkChannelParams as UnlinkChannelParams);
```

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx (hunk 2 above)
Comment (SaurabhSharma-884) [reply webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:114]: Also we can avoid destructive dialog for Retry actions

**Review by SaurabhSharma-884 — CHANGES_REQUESTED** [review]

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx (hunk 6 above)
Comment (ayusht2810) [reply webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:58]: fixed

File: webapp/src/components/Dialog/Dialog.component.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+export const Dialog = ({
+    show,
+    title,
+    destructive,
+    primaryButtonText,
+    secondaryButtonText,
+    onSubmitHandler,
+    onCloseHandler,
+    children,
+    isLoading,
```
Comment (SaurabhSharma-884) [inline webapp/src/components/Dialog/Dialog.component.tsx:14]: Add default values for optional props

**Review by SaurabhSharma-884 — APPROVED** [review]

File: webapp/src/components/Dialog/Dialog.component.tsx [hunk 9]
```diff
@@ -1,43 +1,30 @@
...
+    className,
+}: DialogProps & {isLoading?: boolean, children: React.ReactNode}) => (
+    <MMDialog
+        show={show}
+        destructive={destructive}
+        primaryButtonText={primaryButtonText}
+        secondaryButtonText={secondaryButtonText}
+        onCloseHandler={() => onCloseHandler()}
```
Comment (avas27JTG) [inline webapp/src/components/Dialog/Dialog.component.tsx:22]:
```suggestion
        onCloseHandler={onCloseHandler}
```

File: webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx [hunk 10]
```diff
@@ -1,61 +1,125 @@
...
+            <Dialog
+                show={showRetryDialog}
+                title='Unlink error'
+                destructive={true}
+                primaryButtonText='Try Again'
+                secondaryButtonText='Cancel'
+                onSubmitHandler={unlinkChannel}
+                onCloseHandler={() => setShowRetryDialog(false)}
+                isLoading={isUnlinkChannelsLoading}
+
```
Comment (avas27JTG) [inline webapp/src/components/LinkedChannelCard/LinkedChannelCard.component.tsx:119]:
```suggestion
```

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**